### PR TITLE
Profile API merge error

### PR DESCRIPTION
**Pull request by jace-roell**
**What It Does**
Resolves an issue where team config base profiles were being overwritten if a user config did not have a base profile.

**How to Test**
See attached issue for details.

Example Config Files:

zowe.config.user.json:
```
{
    "$schema": "./zowe.schema.json",
    "profiles": {
        "zosmf": {
            "type": "zosmf",
            "properties": {
                "port": <port>
            }
        }
    },
    "defaults": {},
    "autoStore": true
}
```
zowe.config.json:
```
{
    "$schema": "./zowe.schema.json",
    "profiles": {
        "zosmf":{
            "type": "zosmf",
            "properties": {
                "user": <user>,
                "password": <password>
            }
        },
        "global_base": {
            "type": "base",
            "properties": {
                "host": <host>,
                "rejectUnauthorized": false
            }
        }
    },
    "defaults": {
        "zosmf": "zosmf",
        "tso": "tso",
        "ssh": "ssh",
        "base": "global_base"
    },
    "autoStore": false
}
```
**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Review by traeok — APPROVED** [review]
Changes and tests LGTM - I tested with Zowe Explorer and its working as expected. Thanks for the fix Jace!

Before:
image

After:
image

**Comment by gejohnston** [comment]
@jace-roell A user config merges differently when both config files are at the same level (profiles are merged property-by-property) than when one profile is global and one profile is in a project directory (the entire profile is replaced). Details are documented here:

[zowe-cli/docs/How_config_files_are_merged.md](https://github.com/zowe/zowe-cli/blob/prof-info-api-fail/docs/How_config_files_are_merged.md)

Have you manually tested such combinations to confirm that they all work as intended?

**Review by zFernand0 — DISMISSED** [review]
In practice the properties seems to be loaded in accordance with the document that Geneshared:
- See [using-a-global-team-config-and-a-global-user-config](https://github.com/zowe/zowe-cli/blob/prof-info-api-fail/docs/How_config_files_are_merged.md#using-a-global-team-config-and-a-global-user-config)

<details><summary>Using the script below</summary>

```typescript
/* Save this file as `test.mjs` and run `node test.mjs` */

import { ProfileInfo } from "@zowe/imperative";
import * as util from "util";
(async () => {
    const profInfo = new ProfileInfo("zowe");
    await profInfo.readProfilesFromDisk();
    const zosmfProfAttrs = profInfo.getDefaultProfile("zosmf");
    const zosmfMergedArgs = profInfo.mergeArgsForProfile(zosmfProfAttrs, { getSecureVals: true });
    console.log("Known args: ", util.inspect(zosmfMergedArgs.knownArgs, {depth: 3, colors: true}))
})().catch((err) => {
    console.error(err);
    process.exit(1);
});
```
</details>

I can see the following outputs:
- <details><summary>On `prof-info-api-fail` - rejectUnathorized is merged according to the document</summary>

  ```shell
  zFernand0 @ Mac (arm64) ~/gh/zowe/cli.master                                                                          [15:39:25]
  > $ node test.mjs                                                                            ⬡ system [±prof-info-api-fail ●(✹)]
      Known args:  [
        {
          argName: 'port',
          dataType: 'number',
          argValue: 1234,
          argLoc: {
            locType: 1,
            osLoc: [ '/Users/zFernand0/.zowe/zowe.config.user.json' ],
            jsonLoc: 'profiles.zosmf.properties.port'
          },
          secure: false,
          inSchema: true
        },
        {
          argName: 'rejectUnauthorized',
          dataType: 'boolean',
          argValue: false,
          argLoc: {
            locType: 1,
            osLoc: [ '/Users/zFernand0/.zowe/zowe.config.user.json' ],
            jsonLoc: 'profiles.base.properties.rejectUnauthorized'
          },
          secure: false,
          inSchema: true
        }
      ]
  ```

  </details>
- <details><summary>On Master (rejectUnathorized is not merged according to the document)</summary>

  ```shell
  zFernand0 @ Mac (arm64) ~/gh/zowe/cli.master                                                                          [15:39:25]
  > $ node test.mjs                                                                                        ⬡ system [±master ●(✹)]
      Known args:  [
        {
          argName: 'port',
          dataType: 'number',
          argValue: 1234,
          argLoc: {
            locType: 1,
            osLoc: [ '/Users/zFernand0/.zowe/zowe.config.user.json' ],
            jsonLoc: 'profiles.zosmf.properties.port'
          },
          secure: false,
          inSchema: true
        }
      ]
  ```

  </details>
  
---
  
The reason for the requested changes is that is you look closely, the `osLoc` for rejecjectUnauthorized is pointing to the global user config instead of where it came from (the global regular/non-user config). Having a wrong `osLoc` value may cause other issues in the future if properties need to be updated. I believe we go to that osLoc and update them. Then, attempting to update the rejectUnauthorized value may result in a new base profile being created in the config.user.json file 😋 

---

And yeah, my configs are as simple as:

<details><summary>`zowe.config.user.json`</summary>

```json
{
    "$schema": "./zowe.schema.json",
    "profiles": {
        "zosmf": {
            "type": "zosmf",
            "properties": {
                "port": 1234
            },
            "secure": []
        }
    },
    "defaults": {
        "zosmf": "zosmf"
    },
    "autoStore": true
}
```

</details>

<details><summary>`zowe.config.json`</summary>

```json
{
    "$schema": "./zowe.schema.json",
    "profiles": {
        "base": {
            "type": "base",
            "properties": {
                "rejectUnauthorized": false
            },
            "secure": []
        }
    },
    "defaults": {
        "base": "base"
    },
    "autoStore": true
}
```

<details>

File: packages/imperative/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Imperative package will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Resolved issue where team config base profiles were being overwritten if a user config did not have a base profile. [#2383](https://github.com/zowe/zowe-cli/pull/2383)
```
Comment (anaxceron) [inline packages/imperative/CHANGELOG.md:7]: Just to be more clear, we should say that we mean config files: "Resolved an issue where base profiles in a team configuration file were overwritten when a user configuration file did not include a base profile."

**Review by anaxceron — COMMENTED** [review]
Suggesting some edits to the changelog entry

**Review by anaxceron — APPROVED** [review]
nice changelog @jace-roell!

**Review by zFernand0 — COMMENTED** [review]
There are some lint warnings regarding indentation after removing the if/else login for the `osLocInfo` 😋 
https://github.com/zowe/zowe-cli/actions/runs/12394840626/job/34599196199?pr=2383#step:5:44

But other than that, this PR LGTM! 😋

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**